server/src/server.rs

```diff
@@ -579,6 +600,36 @@ impl Server {
         server.nodes.check_timeouts(SESSION_TIMEOUT);
     }
 
+    async fn forward_resumer(&self) {
+        let mut interval = time::interval(Duration::new(SESSION_CLEANER_INTERVAL, 0));
+        loop {
+            interval.tick().await;
+            let s = self.clone();
+            s.check_resume_forwarding().await;
+        }
+    }
+
+    async fn check_resume_forwarding(&self) {
+        let mut server = self.state.write().await;
+        let clock = DefaultClock::default();
+        let rs_fwd = server.resume_forwarding.clone();
+        let mut set_iter = rs_fwd.iter();
+        while let Some(elem) = set_iter.next() {
+            let (resume_at, session_id, socket_addr) = elem.clone();
+            let now = clock.now();
+            if resume_at > now {
+                break;
+            }
+            server.resume_forwarding.remove(elem);
+            if let Some(node_session) = server.nodes.get_by_session(session_id.clone()) {
+                let control_packet = proto::Packet::control(session_id.to_vec(), ya_relay_proto::proto::control::ResumeForwarding { slot: node_session.info.slot });
+                if let Err(e) = self.send_to(PacketKind::Packet(control_packet), &socket_addr).await {
```

Comment:
Do we know whether this will cause delays or should be spawned in background?

You could acquire the server lock once and won't need to clone `resume_forwarding` instead

Reply:
Fixed the lock issue.
Unfortunately only `.spawn_local()` is available for `Server`.